lib/buckets/common.js

```diff
@@ -74,36 +74,42 @@ function _list(type, req, bucket) {
     var requestId = req.getId();
 
     var owner = req.owner.account.uuid;
-    var sorted = req.params.sorted === 'true';
     var prefix = req.params.prefix || '';
-    var order_by = req.params.sort === 'mtime' ? 'created' : 'name';
+    var marker = req.params.marker || '';
+    var delimiter = req.params.delimiter || '';
+
+    assert.uuid(owner, 'owner');
+    assert.string(prefix, 'prefix');
+    assert.string(marker, 'marker');
 
     var limit;
     if (req.params.limit) {
         limit = parseInt(req.params.limit, 10);
-        if (isNaN(limit) || limit <= 0) {
+        if (isNaN(limit) || limit <= 0 || limit > 1024) {
             process.nextTick(function () {
                 ee.emit('error', new InvalidLimitError(req.params.limit));
             });
             return (ee);
         }
     } else {
-        limit = 0;
+        limit = 1024;
     }
+
     assert.number(limit, 'limit');
-    assert.ok(limit >= 0, 'limit >= 0');
+    assert.ok(limit > 0, 'limit > 0');
+    assert.ok(limit <= 1024, 'limit <= 1024');
 
     log.debug('buckets common _list (%s): entered', type);
 
     var mreq;
     switch (type) {
     case 'buckets':
-        mreq = client.listBucketsNoVnode(owner, sorted, order_by, prefix,
```

Comment:
Do you think it would be possible to jointly support the sorted listing operations that allow the use of the `limit`, `prefix`, `marker`, and `delimiter` and also an unsorted listing that does not honor the options? Maybe they even branch into totally separate rpc calls sent to electric-boray if that helps keep the logic clean. The unsorted use case would be to allow a user to generate a snapshot of all the objects in a bucket when order is not important and doing it unsorted lets us achieve that result without the sorting overhead. This doesn't have to be done as part of this ticket, but just raising it now for consideration.

Reply:
I really don't think so.  The only way I could see to do this would be to allow an unlimited select statement on the database.  If we did support this, the user would only get 1024 results back and not have any feasible way to paginate for new results.  Unless you're only concerned about the first 1024 results? In which case, the sorting overhead wouldn't be that large I imagine.